src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs

```diff
@@ -448,5 +448,134 @@ public async Task DebugHotReloadMethodEmpty()
+                "dotnet://debugger-test.dll/debugger-test.cs",
+                bp_conditional.Value["locations"][0]["lineNumber"].Value<int>(),
+                bp_conditional.Value["locations"][0]["columnNumber"].Value<int>(),
+                "LoopToBreak",
+                locals_fn: (locals) =>
+                {
+                    CheckNumber(locals, "i", 0);
+                }
+            );
+
+            await SendCommandAndCheck(null, "Debugger.resume",
+                null,
+                bp_conditional.Value["locations"][0]["lineNumber"].Value<int>(),
+                bp_conditional.Value["locations"][0]["columnNumber"].Value<int>(),
+                "LoopToBreak",
+                locals_fn: (locals) =>
+                {
+                    CheckNumber(locals, "i", 3);
+                });
+
+            await SendCommandAndCheck(null, "Debugger.resume",
+                null,
+                bp_conditional.Value["locations"][0]["lineNumber"].Value<int>(),
+                bp_conditional.Value["locations"][0]["columnNumber"].Value<int>(),
+                "LoopToBreak",
+                locals_fn: (locals) =>
+                {
+                    CheckNumber(locals, "i", 6);
+                });
+
+            await SendCommandAndCheck(null, "Debugger.resume",
+                null,
+                bp_conditional.Value["locations"][0]["lineNumber"].Value<int>(),
+                bp_conditional.Value["locations"][0]["columnNumber"].Value<int>(),
+                "LoopToBreak",
+                locals_fn: (locals) =>
+                {
+                    CheckNumber(locals, "i", 9);
+                });
+
+            await SendCommandAndCheck(null, "Debugger.resume",
+                null,
+                bp_check.Value["locations"][0]["lineNumber"].Value<int>(),
+                bp_check.Value["locations"][0]["columnNumber"].Value<int>(),
+                "LoopToBreak");
+        }
+
+        [Fact]
+        public async Task ConditionalBreakpointNoStopInALoop()
+        {
+            var bp_conditional = await SetBreakpointInMethod("debugger-test.dll", "LoopClass", "LoopToBreak", 4, condition:"i == \"10\"");
+            var bp_check = await SetBreakpointInMethod("debugger-test.dll", "LoopClass", "LoopToBreak", 5);
+            await EvaluateAndCheck(
+                "window.setTimeout(function() { invoke_static_method('[debugger-test] LoopClass:LoopToBreak'); }, 1);",
+                "dotnet://debugger-test.dll/debugger-test.cs",
+                bp_check.Value["locations"][0]["lineNumber"].Value<int>(),
+                bp_check.Value["locations"][0]["columnNumber"].Value<int>(),
+                "LoopToBreak"
+            );
+        }
+
+        [Fact]
+        public async Task ConditionalBreakpointNotBooleanInALoop()
+        {
+            var bp_conditional = await SetBreakpointInMethod("debugger-test.dll", "LoopClass", "LoopToBreak", 4, condition:"i + 4");
```

Comment:
Should the conditional breakpoint be hit at all? I thought that a non-bool condition would make it evaluate to "false", or silently fail

Reply:
I'm not sure but I think we have already discussed this, when I implemented the other test cases. :)
The behavior is the same that we have for JS, if on JS we evaluate a number > 0 or a string with any value, it will stop in the breakpoint.
As you can see in this test.
https://github.com/thaystg/runtime/blob/f6f398e570a2368309229ce10f24347584813827/src/mono/wasm/debugger/DebuggerTestSuite/BreakpointTests.cs#L221

Reply:
Ah, you are correct. We did discuss -https://github.com/dotnet/runtime/pull/48785#discussion_r584208969 .